Bottom: 55e6b0286566926bbfb72db447075e9f435fad7f
Top:    54a2dc1a2d00cef4f5e3ce3e7793485e412b8038
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-27 14:44:38 +0000

Now it appears to work - YAY....

Turns out that we should NOT attempt to invert the top and bottom
faces of the via barrels, as when we insert them into the opposite
side-sense faces of the copper above and below them, the edge
orientation is already correct without any cheating.

TODO: Reduce operations on the extrusion to positive / negative,
      as it doesn't make sense to leave the multitude of broken
      options.


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index b94327f..9103e5f 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -66,7 +66,7 @@
 
 #define HACK_BOARD_THICKNESS MM_TO_COORD(1.6)
 #define HACK_COPPER_THICKNESS MM_TO_COORD(0.035)
-#define HACK_PLATED_BARREL_THICKNESS MM_TO_COORD(0.02)
+#define HACK_PLATED_BARREL_THICKNESS MM_TO_COORD(0.08) //2
 #define HACK_MASK_THICKNESS MM_TO_COORD(0.01)
 #define HACK_SILK_THICKNESS MM_TO_COORD(0.01)
 
@@ -696,6 +696,7 @@ object3d_from_contours (POLYAREA *contours,
         normal_z = cw ? 1. : -1.; /* NORMAL POINTING TO -VE Z MAKES CIRCLE CLOCKWISE */
 #else
         normal_z = cw ? -1. : 1.; /* NORMAL POINTING TO -VE Z MAKES CIRCLE CLOCKWISE */
+        //normal_z = (top_bot_inverted != cw) ? -1. : 1.; /* NORMAL POINTING TO -VE Z MAKES CIRCLE CLOCKWISE */
 #endif
 
         edge_info_set_round (UNDIR_DATA (edges[i]),
@@ -1886,7 +1887,7 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
 #endif
                                                copper_appearance,
                                                NULL,
-                                               false, true);
+                                               false, false/*true*/);
 
 /* Connect the via barrels in this block of code */
 #if 1
diff --git a/src/hid/step/step_writer.c b/src/hid/step/step_writer.c
index 3ec1b4b..89bbfd4 100644
--- a/src/hid/step/step_writer.c
+++ b/src/hid/step/step_writer.c
@@ -343,7 +343,7 @@ step_face_bound (step_file *file, char *name, step_id bound, bool orientation)
 step_id
 step_face_outer_bound (step_file *file, char *name, step_id bound, bool orientation)
 {
-  fprintf (file->f, "#%i=FACE_OUTER_BOUND('%s',#%i,%s);\n",
+  fprintf (file->f, "#%i=FACE_OUTER_BOUND('%s',#%i,%s);",
                     file->next_id, name, bound, step_bool (orientation));
 
   return file->next_id++;
